File: javascript/VEvent.js


function VEvent(p, pe)
{
	this.point = p;
	this.pe = pe;
	this.y = p.y;
	this.key = Math.random()*100000000;
	
	this.arch = null;
	this.value = 0;
}

VEvent.prototype.compare = function(other)
{
	return((this.y>other.y)?1:-1);
}

File: ported/VEvent.py
__author__ = 'sean'

import random

class VEvent:
    def __init__(self, p, pe):
        self.point = p
        self.pe = pe
        self.y = p.y
        self.key = random.random() * 100000000
        self.arch = None
        self.value = 0
    def compare(self, other):
        #TODO:  need to check if this works
        if(self.y > other.y):
            return True
        else:
            return False